recommend/imageHandler.go

```diff
@@ -0,0 +1,366 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Authors of KubeArmor
+
+package recommend
+
+import (
+	"archive/tar"
+	"bufio"
+	"context"
+	_ "embed" // need for embedding
+	"encoding/base64"
+	"fmt"
+	"io"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/clarketm/json"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/docker/docker/pkg/jsonmessage"
+	"github.com/fatih/color"
+	"github.com/moby/term"
+	log "github.com/sirupsen/logrus"
+)
+
+var cli *client.Client // docker client
+var tempDir string     // temporary directory used by karmor to save image etc
+
+// ImageInfo contains image information
+type ImageInfo struct {
+	Name     string
+	RepoTags []string
+	Arch     string
+	Distro   string
+	OS       string
+	FileList []string
+	DirList  []string
+}
+
+func getAuthStr() string {
+	u := os.Getenv("DOCKER_USERNAME")
+	p := os.Getenv("DOCKER_PASSWORD")
+	if u == "" || p == "" {
+		return ""
+	}
```

Comment:
What do we need these for? Private images?

Reply:
Yep, for private images. I thought it would be better to have atleast some option to specify username/token